Gradle build fails after upgrade to 1.7.2

After upgrade to 1.7.2, all maven projects work just fine, but gradle projects fail with:
```
Step #0 - "Build and Push Image": > Task :quarkusGenerateCode
Step #0 - "Build and Push Image": preparing quarkus application
Step #0 - "Build and Push Image": kotlin scripting plugin: root project 'my-service'.compileQuarkus-generated-sourcesKotlin - configuration not found: quarkus-generated-sourcesKotlinScriptDef, the plugin is probably applied by a mistake
Step #0 - "Build and Push Image": 
Step #0 - "Build and Push Image": > Task :quarkusGenerateCodeTests FAILED
Step #0 - "Build and Push Image": preparing quarkus application
Step #0 - "Build and Push Image": kotlin scripting plugin: root project 'my-service'.compileQuarkus-test-generated-sourcesKotlin - configuration not found: quarkus-test-generated-sourcesKotlinScriptDef, the plugin is probably applied by a mistake
Step #0 - "Build and Push Image": 
Step #0 - "Build and Push Image": FAILURE: Build failed with an exception.
Step #0 - "Build and Push Image": 
Step #0 - "Build and Push Image": * What went wrong:
Step #0 - "Build and Push Image": Execution failed for task ':quarkusGenerateCodeTests'.
Step #0 - "Build and Push Image": > Failed to generate sources in the QuarkusPrepare task
```

Any thoughts about why it happened and how to fix that?

This is what I see with `--stacktrace`: 

```java
Caused by: java.lang.NullPointerException
        at io.quarkus.deployment.CodeGenerator.codeGenOutDir(CodeGenerator.java:99)
        at io.quarkus.deployment.CodeGenerator.lambda$init$0(CodeGenerator.java:49)
        at io.quarkus.deployment.CodeGenerator.callWithClassloader(CodeGenerator.java:65)
        at io.quarkus.deployment.CodeGenerator.init(CodeGenerator.java:38)
        at io.quarkus.deployment.CodeGenerator.initAndRun(CodeGenerator.java:27)
        ... 97 more
```

--- a/devtools/gradle/src/main/java/io/quarkus/gradle/QuarkusPlugin.java
+++ b/devtools/gradle/src/main/java/io/quarkus/gradle/QuarkusPlugin.java
@@ -139,13 +139,9 @@ public class QuarkusPlugin implements Plugin<Project> {
                     compileJavaTask.dependsOn(quarkusGenerateCode);
                     quarkusGenerateCode.setSourceRegistrar(compileJavaTask::source);
 
-                    Task compileKotlinTask = tasks.findByName("compileKotlin");
-                    // TODO: proper support of kotlin
-                    if (compileKotlinTask == null) {
-                        JavaCompile compileTestJavaTask = (JavaCompile) tasks.getByName(JavaPlugin.COMPILE_TEST_JAVA_TASK_NAME);
-                        compileTestJavaTask.dependsOn(quarkusGenerateCodeTests);
-                        quarkusGenerateCodeTests.setSourceRegistrar(compileTestJavaTask::source);
-                    }
+                    JavaCompile compileTestJavaTask = (JavaCompile) tasks.getByName(JavaPlugin.COMPILE_TEST_JAVA_TASK_NAME);
+                    compileTestJavaTask.dependsOn(quarkusGenerateCodeTests);
+                    quarkusGenerateCodeTests.setSourceRegistrar(compileTestJavaTask::source);
 
                     Task classesTask = tasks.getByName(JavaPlugin.CLASSES_TASK_NAME);
                     Task resourcesTask = tasks.getByName(JavaPlugin.PROCESS_RESOURCES_TASK_NAME);
@@ -187,7 +183,7 @@ public class QuarkusPlugin implements Plugin<Project> {
                     tasks.withType(Test.class).whenTaskAdded(configureTestTask::accept);
                 });
         project.getPlugins().withId("org.jetbrains.kotlin.jvm", plugin -> {
-            Task compileKotlinTask = tasks.findByName("compileKotlin");
+            Task compileKotlinTask = tasks.getByName("compileKotlin");
             compileKotlinTask.dependsOn(quarkusGenerateCode, quarkusGenerateCodeTests);
         });
     }

--- a/integration-tests/gradle/src/test/java/io/quarkus/gradle/KotlinGRPCProjectBuildTest.java
+++ b/integration-tests/gradle/src/test/java/io/quarkus/gradle/KotlinGRPCProjectBuildTest.java
@@ -1,0 +1,18 @@
+package io.quarkus.gradle;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.io.File;
+
+import org.junit.jupiter.api.Test;
+
+public class KotlinGRPCProjectBuildTest extends QuarkusGradleWrapperTestBase {
+
+    @Test
+    public void testBasicMultiModuleBuild() throws Exception {
+        final File projectDir = getProjectDir("kotlin-grpc-project");
+        final BuildResult build = runGradleWrapper(projectDir, "clean", "build");
+        assertThat(build.getTasks().get(":quarkusGenerateCode")).isEqualTo(BuildResult.SUCCESS_OUTCOME);
+        assertThat(build.getTasks().get(":compileKotlin")).isEqualTo(BuildResult.SUCCESS_OUTCOME);
+    }
+}

--- a/integration-tests/gradle/src/test/resources/kotlin-grpc-project/build.gradle
+++ b/integration-tests/gradle/src/test/resources/kotlin-grpc-project/build.gradle
@@ -1,0 +1,55 @@
+plugins {
+    id 'org.jetbrains.kotlin.jvm' version "1.3.72"
+    id "org.jetbrains.kotlin.plugin.allopen" version "1.3.72"
+    id 'io.quarkus'
+}
+
+repositories {
+     mavenLocal()
+     mavenCentral()
+}
+
+dependencies {
+    implementation 'io.quarkus:quarkus-kotlin'
+    implementation 'io.quarkus:quarkus-grpc'
+    implementation enforcedPlatform("${quarkusPlatformGroupId}:${quarkusPlatformArtifactId}:${quarkusPlatformVersion}")
+    implementation 'io.quarkus:quarkus-resteasy'
+    implementation 'org.jetbrains.kotlin:kotlin-stdlib-jdk8'
+
+    testImplementation 'io.quarkus:quarkus-junit5'
+    testImplementation 'io.rest-assured:kotlin-extensions'
+}
+
+group 'org.acme'
+version '1.0.0-SNAPSHOT'
+
+quarkus {
+    setOutputDirectory("$projectDir/build/classes/kotlin/main")
+}
+
+quarkusDev {
+    setSourceDir("$projectDir/src/main/kotlin")
+}
+
+allOpen {
+    annotation("javax.ws.rs.Path")
+    annotation("javax.enterprise.context.ApplicationScoped")
+    annotation("io.quarkus.test.junit.QuarkusTest")
+}
+
+java {
+    sourceCompatibility = JavaVersion.VERSION_11
+    targetCompatibility = JavaVersion.VERSION_11
+}
+
+compileKotlin {
+    kotlinOptions.jvmTarget = JavaVersion.VERSION_11
+    kotlinOptions.javaParameters = true
+}
+
+compileTestKotlin {
+    kotlinOptions.jvmTarget = JavaVersion.VERSION_11
+}
+test {
+    systemProperty "java.util.logging.manager", "org.jboss.logmanager.LogManager"
+}

--- a/integration-tests/gradle/src/test/resources/kotlin-grpc-project/gradle.properties
+++ b/integration-tests/gradle/src/test/resources/kotlin-grpc-project/gradle.properties
@@ -1,0 +1,2 @@
+quarkusPlatformArtifactId=quarkus-bom
+quarkusPlatformGroupId=io.quarkus

--- a/integration-tests/gradle/src/test/resources/kotlin-grpc-project/settings.gradle
+++ b/integration-tests/gradle/src/test/resources/kotlin-grpc-project/settings.gradle
@@ -1,0 +1,11 @@
+pluginManagement {
+    repositories {
+        mavenLocal()
+        mavenCentral()
+        gradlePluginPortal()
+    }
+    plugins {
+      id 'io.quarkus' version "${quarkusPluginVersion}"
+    }
+}
+rootProject.name='code-with-quarkus'

--- a/integration-tests/gradle/src/test/resources/kotlin-grpc-project/src/main/kotlin/org/acme/ExampleResource.kt
+++ b/integration-tests/gradle/src/test/resources/kotlin-grpc-project/src/main/kotlin/org/acme/ExampleResource.kt
@@ -1,0 +1,14 @@
+package org.acme
+
+import javax.ws.rs.GET
+import javax.ws.rs.Path
+import javax.ws.rs.Produces
+import javax.ws.rs.core.MediaType
+
+@Path("/hello")
+class ExampleResource {
+
+    @GET
+    @Produces(MediaType.TEXT_PLAIN)
+    fun hello() = "hello"
+}

--- a/integration-tests/gradle/src/test/resources/kotlin-grpc-project/src/main/resources/META-INF/resources/index.html
+++ b/integration-tests/gradle/src/test/resources/kotlin-grpc-project/src/main/resources/META-INF/resources/index.html
@@ -1,0 +1,155 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <title>code-with-quarkus - 1.0.0-SNAPSHOT</title>
+    <style>
+        h1, h2, h3, h4, h5, h6 {
+            margin-bottom: 0.5rem;
+            font-weight: 400;
+            line-height: 1.5;
+        }
+
+        h1 {
+            font-size: 2.5rem;
+        }
+
+        h2 {
+            font-size: 2rem
+        }
+
+        h3 {
+            font-size: 1.75rem
+        }
+
+        h4 {
+            font-size: 1.5rem
+        }
+
+        h5 {
+            font-size: 1.25rem
+        }
+
+        h6 {
+            font-size: 1rem
+        }
+
+        .lead {
+            font-weight: 300;
+            font-size: 2rem;
+        }
+
+        .banner {
+            font-size: 2.7rem;
+            margin: 0;
+            padding: 2rem 1rem;
+            background-color: #00A1E2;
+            color: white;
+        }
+
+        body {
+            margin: 0;
+            font-family: -apple-system, system-ui, "Segoe UI", Roboto, "Helvetica Neue", Arial, sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol", "Noto Color Emoji";
+        }
+
+        code {
+            font-family: SFMono-Regular, Menlo, Monaco, Consolas, "Liberation Mono", "Courier New", monospace;
+            font-size: 87.5%;
+            color: #e83e8c;
+            word-break: break-word;
+        }
+
+        .left-column {
+            padding: .75rem;
+            max-width: 75%;
+            min-width: 55%;
+        }
+
+        .right-column {
+            padding: .75rem;
+            max-width: 25%;
+        }
+
+        .container {
+            display: flex;
+            width: 100%;
+        }
+
+        li {
+            margin: 0.75rem;
+        }
+
+        .right-section {
+            margin-left: 1rem;
+            padding-left: 0.5rem;
+        }
+
+        .right-section h3 {
+            padding-top: 0;
+            font-weight: 200;
+        }
+
+        .right-section ul {
+            border-left: 0.3rem solid #00A1E2;
+            list-style-type: none;
+            padding-left: 0;
+        }
+
+    </style>
+</head>
+<body>
+
+<div class="banner lead">
+    Your new Cloud-Native application is ready!
+</div>
+
+<div class="container">
+    <div class="left-column">
+        <p class="lead"> Congratulations, you have created a new Quarkus application.</p>
+
+        <h2>Why do you see this?</h2>
+
+        <p>This page is served by Quarkus. The source is in
+            <code>src/main/resources/META-INF/resources/index.html</code>.</p>
+
+        <h2>What can I do from here?</h2>
+
+        <p>If not already done, run the application in <em>dev mode</em> using: <code>mvn compile quarkus:dev</code>.
+        </p>
+        <ul>
+            <li>Add REST resources, Servlets, functions and other services in <code>src/main/java</code>.</li>
+            <li>Your static assets are located in <code>src/main/resources/META-INF/resources</code>.</li>
+            <li>Configure your application in <code>src/main/resources/application.properties</code>.
+            </li>
+        </ul>
+
+        <h2>Do you like Quarkus?</h2>
+        <p>Go give it a star on <a href="https://github.com/quarkusio/quarkus">GitHub</a>.</p>
+
+        <h2>How do I get rid of this page?</h2>
+        <p>Just delete the <code>src/main/resources/META-INF/resources/index.html</code> file.</p>
+    </div>
+    <div class="right-column">
+        <div class="right-section">
+            <h3>Application</h3>
+            <ul>
+                <li>GroupId: org.acme</li>
+                <li>ArtifactId: code-with-quarkus</li>
+                <li>Version: 1.0.0-SNAPSHOT</li>
+                <li>Quarkus Version: 1.7.2.Final</li>
+            </ul>
+        </div>
+        <div class="right-section">
+            <h3>Next steps</h3>
+            <ul>
+                <li><a href="https://quarkus.io/guides/maven-tooling.html" target="_blank">Setup your IDE</a></li>
+                <li><a href="https://quarkus.io/guides/getting-started.html" target="_blank">Getting started</a></li>
+                <li><a href="https://quarkus.io" target="_blank">Quarkus Web Site</a></li>
+            </ul>
+        </div>
+    </div>
+</div>
+
+
+</body>
+</html>

--- a/integration-tests/gradle/src/test/resources/kotlin-grpc-project/src/main/resources/application.properties
+++ b/integration-tests/gradle/src/test/resources/kotlin-grpc-project/src/main/resources/application.properties
@@ -1,0 +1,2 @@
+# Configuration file
+# key = value

--- a/integration-tests/gradle/src/test/resources/kotlin-grpc-project/src/native-test/kotlin/org/acme/NativeExampleResourceIT.kt
+++ b/integration-tests/gradle/src/test/resources/kotlin-grpc-project/src/native-test/kotlin/org/acme/NativeExampleResourceIT.kt
@@ -1,0 +1,6 @@
+package org.acme
+
+import io.quarkus.test.junit.NativeImageTest
+
+@NativeImageTest
+class NativeExampleResourceIT : ExampleResourceTest()

--- a/integration-tests/gradle/src/test/resources/kotlin-grpc-project/src/test/kotlin/org/acme/ExampleResourceTest.kt
+++ b/integration-tests/gradle/src/test/resources/kotlin-grpc-project/src/test/kotlin/org/acme/ExampleResourceTest.kt
@@ -1,0 +1,20 @@
+package org.acme
+
+import io.quarkus.test.junit.QuarkusTest
+import io.restassured.RestAssured.given
+import org.hamcrest.CoreMatchers.`is`
+import org.junit.jupiter.api.Test
+
+@QuarkusTest
+class ExampleResourceTest {
+
+    @Test
+    fun testHelloEndpoint() {
+        given()
+          .`when`().get("/hello")
+          .then()
+             .statusCode(200)
+             .body(`is`("hello"))
+    }
+
+}

